### PR TITLE
Tweak documentation on styling selections and link from Tools

### LINKED ISSUE #2117: Mention nonselection_glyph and selection_glyph options in Tools documentation.
(plan to do this, just noting for now)

In each documentation for a select tool, where there is already:

"Selections can be comprised of multiple regions, even those
made by different selection tools. Hold down the <<shift>> key
while making a selection to append the new selection to any
previous seletion that might exist."
e.g. https://github.com/bokeh/bokeh/blob/master/bokeh/models/tools.py#L150

Add a note that you can customize the display of the selected and unselected glyphs by passing selection_glyph and nonselection_glyph parameter to add_glyph method.

### PR DESCRIPTION
**Pull request by birdsarah**
issues: fixes #2117

### CODE REVIEW

File: sphinx/source/docs/user_guide/source_examples/styling_glyph_selections.py [hunk 1]
```diff
@@ -1,14 +1,18 @@
...
-glyph.line_color = "firebrick"
-glyph.line_dash = [6, 3]
-glyph.line_width = 2
+selected_circle = Circle(fill_alpha=1, fill_color="firebrick", line_color=None)
+nonselected_circle = Circle(fill_alpha=0.2, fill_color="blue", line_color="firebrick")
+
+circle_glyph_renderer = p.select(name="mycircle")[0]
+
+circle_glyph_renderer.selection_glyph = selected_circle
+circle_glyph_renderer.nonselection_glyph = nonselected_circle
```
Comment (bryevdv) [inline sphinx/source/docs/user_guide/source_examples/styling_glyph_selections.py]: This works without the `[0]` also I'd suggest just `renderer`, and a title that explicitly says to click:
```
from bokeh.io import output_file, show
from bokeh.plotting import figure
from bokeh.models import Circle

output_file("styling_selections.html")

p = figure(plot_width=400, plot_height=400, tools="tap", title="Select by Clicking")
p.circle([1, 2, 3, 4, 5], [2, 5, 8, 2, 7], size=50, name="mycircle")

selected_circle = Circle(fill_alpha=1, fill_color="firebrick", line_color=None)
nonselected_circle = Circle(fill_alpha=0.2, fill_color="blue", line_color="firebrick")

renderer = p.select(name="mycircle")

renderer.selection_glyph = selected_circle
renderer.nonselection_glyph = nonselected_circle

show(p)
```

File: sphinx/source/docs/user_guide/styling.rst [hunk 2]
```diff
@@ -252,21 +250,54 @@ This is the object to set fill, line, or text property values for:
...
-and ``nonselection_glyph`` attributes that control the visual appearance of
-glyphs when selection tools are used.
+.. _userguide_styling_selected_unselected_glyphs:
+
+Selected & Unselected Glyphs
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The styling of selected and non-selected glyphs can be customized by
+setting the |selection_glyph| and/or |nonselection_glyph| attributes
+of the |GlyphRenderer| either manually or be passing them to |add_glyph|.
```
Comment (bryevdv) [inline sphinx/source/docs/user_guide/styling.rst]: "by passing"

File: sphinx/source/docs/user_guide/styling.rst [hunk 3]
```diff
@@ -252,21 +250,54 @@ This is the object to set fill, line, or text property values for:
...
-``GlyphRenderer`` objects can also be configured with ``selection_glyph``
-and ``nonselection_glyph`` attributes that control the visual appearance of
-glyphs when selection tools are used.
+.. _userguide_styling_selected_unselected_glyphs:
+
+Selected & Unselected Glyphs
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The styling of selected and non-selected glyphs can be customized by
+setting the |selection_glyph| and/or |nonselection_glyph| attributes
```
Comment (bryevdv) [inline sphinx/source/docs/user_guide/styling.rst]: I'm averse to "and/or" in general, maybe there is a re-wording?

**Comment by birdsarah** [comment]
Thanks for taking the time @bryevdv, much appreciated.

**Comment by bryevdv** [comment]
Thank you, for taking time on the docs!

**Pull request merged**